From d21b0ab5d96f830404d8bd3fad7e6d452d62ef90 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:47 -0400
Subject: [ethernet] cxgb4: Export symbols required by cxgb4i for ipv6 support and required defines

Message-id: <1410475447-94392-89-git-send-email-svemuri@redhat.com>
Patchwork-id: 93818
O-Subject: [RHEL7.1 PATCH BZ 1124947 088/108] cxgb4: Export symbols required by cxgb4i for ipv6 support and required defines
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Anish Bhatt <anish@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit a3e3b2857d35988819bc396c012c53898b8223e6)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 7697871..0aedfb4 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -3280,8 +3280,8 @@ static int tid_init(struct tid_info *t)
  return 0;
 }
 
-static int cxgb4_clip_get(const struct net_device *dev,
-     const struct in6_addr *lip)
+int cxgb4_clip_get(const struct net_device *dev,
+     const struct in6_addr *lip)
 {
  struct adapter *adap;
  struct fw_clip_cmd c;
@@ -3295,9 +3295,10 @@ static int cxgb4_clip_get(const struct net_device *dev,
  c.ip_lo = *(__be64 *)(lip->s6_addr + 8);
  return t4_wr_mbox_meat(adap, adap->mbox, &c, sizeof(c), &c, false);
 }
+EXPORT_SYMBOL(cxgb4_clip_get);
 
-static int cxgb4_clip_release(const struct net_device *dev,
-         const struct in6_addr *lip)
+int cxgb4_clip_release(const struct net_device *dev,
+         const struct in6_addr *lip)
 {
  struct adapter *adap;
  struct fw_clip_cmd c;
@@ -3311,6 +3312,7 @@ static int cxgb4_clip_release(const struct net_device *dev,
  c.ip_lo = *(__be64 *)(lip->s6_addr + 8);
  return t4_wr_mbox_meat(adap, adap->mbox, &c, sizeof(c), &c, false);
 }
+EXPORT_SYMBOL(cxgb4_clip_release);
 
 /**
  * cxgb4_create_server - create an IP server
diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_uld.h b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_uld.h
index ea21182..ad02de0 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_uld.h
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_uld.h
@@ -172,6 +172,10 @@ int cxgb4_create_server_filter(const struct net_device *dev, unsigned int stid,
           unsigned char port, unsigned char mask);
 int cxgb4_remove_server_filter(const struct net_device *dev, unsigned int stid,
           unsigned int queue, bool ipv6);
+int cxgb4_clip_get(const struct net_device *dev, const struct in6_addr *lip);
+int cxgb4_clip_release(const struct net_device *dev,
+         const struct in6_addr *lip);
+
 static inline void set_wr_txq(struct sk_buff *skb, int prio, int queue)
 {
  skb_set_queue_mapping(skb, (queue << 1) | prio);
diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h b/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h
index 65983c7..a9d164b 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_msg.h
@@ -270,6 +270,8 @@ struct cpl_pass_accept_rpl {
 #define RX_COALESCE_VALID(x) ((x) << 11)
 #define RX_COALESCE(x)       ((x) << 12)
 #define PACE(x)       ((x) << 16)
+#define RX_FC_VALID      ((1U) << 19)
+#define RX_FC_DISABLE      ((1U) << 20)
 #define TX_QUEUE(x)          ((x) << 23)
 #define RX_CHANNEL(x)        ((x) << 26)
 #define CCTRL_ECN(x)         ((x) << 27)
-- 
1.7.1